lib/src/internal/FlowManager.cpp

```diff
@@ -152,7 +163,17 @@ namespace mxl::lib
                 gInfo.deviceIndex = -1;
             }
 
-            publishFlowDirectory(tempDirectory, makeFlowDirectoryName(_mxlDomain, uuidString));
+            auto const finalDir = makeFlowDirectoryName(_mxlDomain, uuidString);
+            try
+            {
+                publishFlowDirectory(tempDirectory, finalDir);
+            }
+            catch (std::exception const& e)
+            {
+                MXL_ERROR("FlowManager: Failed to publish flow directory from '{}' to '{}': {}", tempDirectory.string(), finalDir.string(), e.what());
+                throw std::filesystem::filesystem_error(
+                    "FlowManager: Failed to publish flow directory", finalDir, std::make_error_code(std::errc::io_error));
```

Comment:
I don't like swallowing the original exception in favor of a generic `std::filesystem_error`.

If you want to add a log statement in this case this is fine by me, but then I would choose a pattern like this:

```cpp
try
{
    // ...
}
catch (std::exception const& e)
{
    MXL_ERROR("Failed to publish flow directory from '{}' to '{}': {}", tempDirectory.string(), finalDir.string(), e.what());
    throw;
}
catch (...)
{
    MXL_ERROR("Failed to publish flow directory from '{}' to '{}': {}", tempDirectory.string(), finalDir.string(), "An unknown error occurred.");
    throw;
}
```